File: CVS_ObjectDetectorSample_ObjC/CVS_ObjectDetectorSample/BoundingBoxLabelLayer.m
// Copyright (c) Microsoft Corporation.  All rights reserved.

#import "BoundingBoxLabelLayer.h"

@implementation BoundingBoxLabelLayer

- (void)setPadding:(CGSize)padding
{
    _padding = padding;
    [self setNeedsLayout];
}

- (CGSize)preferredFrameSize
{
    CGSize textSize = super.preferredFrameSize;
    return CGSizeMake(
        textSize.width + self.padding.width,
        textSize.height + self.padding.height);
}

- (void)drawInContext:(CGContextRef)ctx
{
    CGContextSaveGState(ctx);
    CGContextTranslateCTM(ctx, self.padding.width / 2.0f, self.padding.height / 2.0f);
    [super drawInContext:ctx];
    CGContextRestoreGState(ctx);
}

@end

File: CVS_ObjectDetectorSample_Swift/CVS_ObjectDetectorSample/BoundingBoxLabelLayer.swift
// Copyright (c) Microsoft Corporation.  All rights reserved.

import UIKit

class BoundingBoxLabelLayer: CATextLayer {

    public var padding = CGSize.zero {
        didSet {
            needsLayout()
        }
    }

    override func preferredFrameSize() -> CGSize {

        let textSize = super.preferredFrameSize();
        return CGSize.init(
            width: textSize.width + padding.width,
            height: textSize.height + padding.height);
    }
    
    override func draw(in ctx: CGContext) {

        ctx.saveGState()
        ctx.translateBy(x: padding.width / 2.0, y: padding.height / 2.0)
        super.draw(in: ctx)
        ctx.restoreGState()
    }
}
